types/networks/src/lib.rs

```diff
@@ -80,6 +97,27 @@ pub fn get_network_version_default(epoch: ChainEpoch) -> NetworkVersion {
         .unwrap_or(NetworkVersion::V0)
 }
 
+/// If true, then we should perform a state migration at this epoch, otherwise, no migration.
+pub fn is_migrate_epoch(epoch: ChainEpoch) -> bool {
+    VERSION_SCHEDULE
+        .iter()
+        .rev()
+        .find(|upgrade| epoch == upgrade.height)
+        .map(|upgrade| upgrade.migration)
+        .unwrap_or(false)
+}
+
+/// Calls the state migration functions to migrate to the new StateTree.
+/// Currently this is a TODO because we havent implemented and migrations yet.
+/// The first migration to implement is from V2 to V3 (Network Version 10), so this method
+/// signature will probably change as we discover more things we need to pass in + error handling.
+pub fn migrate_state<T>(_bs: &T, _old_state: Cid, _epoch: ChainEpoch) -> Cid
```

Comment:
```suggestion
pub fn migrate_state<T>(_bs: &T, _old_state: &Cid, _epoch: ChainEpoch) -> Result<Cid, Box<dyn StdError>>
```

Could wait to change to error prone until it's more clear what errors can arise, but also could wrap `Cid` in an option if you decide to take previous comment.

Reply:
Good idea